main.py

```diff
@@ -336,27 +347,197 @@ def performEvaluation(issue_data) -> Result:
     url = issue_data[JsonKeys.URL.value]
     branch = issue_data[JsonKeys.BRANCH.value]
     commit_hash = issue_data[JsonKeys.COMMIT_HASH.value]
+    qual_jar_required = issue_data[JsonKeys.CHECKER_QUAL_REQURIED.value]
+    qual_jar_dir = ""
 
     issue_folder_abs_dir = os.path.abspath(issue_folder_dir)
     input_dir = create_issue_directory(issue_folder_abs_dir, issue_id)
 
     get_target_data(url, branch, commit_hash, input_dir) 
     
     repo_name = get_repository_name(url)
+    if qual_jar_required:
+        qual_jar_dir = os.path.join(issue_folder_abs_dir, issue_id, specimin_input, repo_name, specimin_project_name)
     specimin_command = ""
     result: Result = None
     specimin_path = get_specimin_env_var()
+    specimin_command = build_specimin_command(repo_name, os.path.join(issue_folder_abs_dir, issue_id), issue_data[JsonKeys.ROOT_DIR.value], issue_data[JsonKeys.TARGETS.value], qual_jar_dir if os.path.exists(qual_jar_dir) else "")
+    
     if specimin_path is not None and os.path.exists(specimin_path):
-        specimin_command = build_specimin_command(repo_name, os.path.join(issue_folder_abs_dir, issue_id), issue_data[JsonKeys.ROOT_DIR.value], issue_data[JsonKeys.TARGETS.value])
         result = run_specimin(issue_id ,specimin_command, specimin_path)
     else:
-        specimin_command = build_specimin_command(repo_name, os.path.join(issue_folder_abs_dir, issue_id),issue_data[JsonKeys.ROOT_DIR.value], issue_data[JsonKeys.TARGETS.value])
         result = run_specimin(issue_id ,specimin_command, os.path.join(issue_folder_abs_dir, specimin_project_name))
     
     print(f"{result.name} - {result.status}")
+
+    # build script is shipped with input program. It exists in the "specimin" directory of the input program's root directory.
+    # Coping the build script to the output directory of the minimized program.
+    build_script_path = os.path.join(issue_folder_abs_dir, issue_id, specimin_input, repo_name, specimin_project_name, "build.gradle")
+
+    if not os.path.exists(build_script_path):
+        return result
```

Comment:
This should raise some kind of exception, should it not?

Reply:
Every target should have a build script. But I have not finished added build script for all. So if the build script is not present, I am just returning the result of minimization result of the target. 

After completion this should raise some flag to indicate that script is missing.